system-helper/flatpak-system-helper.c

```diff
@@ -247,6 +360,33 @@ handle_deploy (FlatpakSystemHelper   *object,
       return TRUE;
     }
 
+  src_dir = g_path_get_dirname (arg_repo_path);
+  ongoing_pull = take_ongoing_pull_by_dir (src_dir);
+  if (ongoing_pull != NULL)
+    {
+      g_autoptr(GError) local_error = NULL;
```

Comment:
I think we need to verify here that the outgoing pull was for the same uid as the caller.

Reply:
Addressed.